.github/workflows/main.yml

```diff
@@ -67,71 +70,130 @@ jobs:
-        echo \
-          "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
-          $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" | \
-          sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
-          sudo apt-get update
-        echo "*****  Check available docker-ce versions ."
-        sudo apt policy docker-ce
-        echo "***** Installing Docker packages..."
-        sudo apt-get install docker-ce=${{ env.DOCKER_APT_PKG_VER }} docker-ce-cli=${{ env.DOCKER_APT_PKG_VER }} containerd.io docker-buildx-plugin docker-compose-plugin
-        echo "***** Verifying initial Docker installation..."
-        docker run hello-world
-        echo "***** Displaying Docker information..."
-        docker info
-        echo "***** Configuring Docker for containerd image store and builder keepStorage..."
-        echo "{ \"features\": { \"containerd-snapshotter\": true }}" | sudo tee /etc/docker/daemon.json
-        echo $'{
-          "features": {
-            "containerd-snapshotter": true
-          },
-          "builder": {
-            "gc": {
-              "defaultKeepStorage": "20GB",
-              "enabled": true
-            }
-          }
-        }' | sudo tee /etc/docker/daemon.json
-        sudo systemctl restart docker
-        docker info -f '{{ .DriverStatus }}'
-
-    - name: Load binfmt platforms for QEMU
+        ./official-images/test/run.sh -c ./official-images/test/config.sh -c test/postgis-config.sh ${{ steps.build.outputs.imageid }}
+
+    - name: Login to dockerhub
+      id: login-dockerhub
+      uses: docker/login-action@v3
+      if: ${{ (github.repository == env.GITHUB_REPO) && (github.ref == 'refs/heads/master') && (github.event_name != 'pull_request') }}
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_ACCESS_TOKEN }}
+
+    - name: Push image by digest
+      id: push
+      uses: docker/build-push-action@v5  # Build is cached, this is really just a push
+      if: ${{ (github.repository == env.GITHUB_REPO) && (github.ref == 'refs/heads/master') && (github.event_name != 'pull_request') }}
+      with:
+        context: ${{ env.VERSION }}${{ env.VARIANT == 'alpine' && '/alpine' || ''}}
+        file: ${{ env.VERSION }}${{ env.VARIANT == 'alpine' && '/alpine' || ''}}/Dockerfile
+        outputs: type=image,"name=${{ env.DOCKERHUB_REPO }}",push-by-digest=true,name-canonical=true,push=true
+
+    - name: Export digest
       run: |
-        docker run --privileged --rm tonistiigi/binfmt --install all
-        docker images --tree
+        mkdir -p ${{ runner.temp }}/digests
+        digest="${{ steps.push.outputs.digest }}"
+        touch "${{ runner.temp }}/digests/${digest#sha256:}"
 
-    - name: Checkout source
-      uses: actions/checkout@v4
+    - name: Upload digests
+      uses: actions/upload-artifact@v4
```

Comment:
imho: The workflow must never push or publish any artifacts, manifests, or Docker images during a Pull Request test run.

so please add : 
`if: ${{ (github.repository == env.GITHUB_REPO) && (github.ref == 'refs/heads/master') && (github.event_name != 'pull_request') }}`

Reply:
Hmmm, this may be the gotcha I ran into before but forgot about.  I'd have to come up to speed again but I remember something about attempting to adhere to the "no registery pushes during PRs" heuristic conflicting with "Need to do a non multiplatform image push to the registry first in order for the multiplatform image manifest creation to be possible" 😬 (but I may be wrong)

Reply:
> Do you care to have both architectures build on PR, or is either fine? If I recall, the current process doesn't yield anything someone can download, so either should be fine unless there's concern that something could break on only one arch.

I'm not sure if this answers your question, but ideally we'd want to have a PR trigger CI to build x86 image on an x86 runner and arm64 image on an arm64 runner (so that both builds happen natively from their own perspective) but the architecture of the runner that handles building the multiplatform manifest and potentially pushing the images/manifests to the registry is probably less of a concern.